packages/blade/src/components/Tour/_decisions/decisions.md

```diff
@@ -0,0 +1,718 @@
+
+**Conclusion:**
+
+- Now that we decided to go with the `TourStep` approach on both web & native, we will have to wrap the whole app with the `Tour` component on both web & native.
+
+## Multiple Tour Flows
+
+In a given product, there can be multiple tour flows, for example:
+
+- Global app level tour (A tour for the whole app, sidebar, search, header etc)
+- And another for example: PaymentLinks page.
+
+Now, the open question is:
+
+1. How will we have two different tour flows in the same page?
+
+This is possible with the current API, but consumers will have to prefix their `id` props manually to avoid conflicts.
+
+### Approach-1: Prefix `id` to avoid conflicts
+
+It will look like this:
+
+```jsx
+
+import { Tour } from '@razorpay/blade/components';
+import type { TourSteps } from '@razorpay/blade/components';
+
+const globalSteps: TourSteps = [
+  {
+    id: 'global-step-1',
+    title: 'Step 1',
+    content: // ...
+  },
+  {
+    id: 'step-2',
+    title: 'Step 2',
+    content: // ...
+  },
+];
+
+const paymentLinkSteps: TourSteps = [
+  {
+    id: 'paymentlink-step-1',
+    title: 'Step 1',
+    content: // ...
+  },
+  {
+    id: 'paymentlink-step-2',
+    title: 'Step 2',
+    content: // ...
+  },
+];
+
+const App = () => {
+  return (
+    <Box>
+      <Tour
+        steps={globalSteps}
+      />
+       <Tour
+        steps={paymentLinksSteps}
+      />
+    </Box>
+  );
+};
+
+// file: DashboardPage.tsx
+const DashboardPage = () => {
+  return (
+    <Box>
+      <Button id="global-step-1">Click me</Button>
+      <Box id="global-step-2" padding="spacing.5">
+        <Text>Some content</Text>
+      </Box>
+    </Box>
+  );
+};
+
+// file: PaymentLinks.tsx
+const PaymentLinksPage = () => {
+  return (
+    <Box>
+      <Button id="paymentlink-step-1">Click me</Button>
+      <Box id="paymentlink-step-2" padding="spacing.5">
+        <Text>Some content</Text>
+      </Box>
+    </Box>
+  );
+};
+```
+
+### Approach-2: Use `TourStep` to avoid conflicts
+
+Now with this approach the `TourStep` component becomes a bit more useful on web too.
+
+What we can do is add a key prop to the `TourStep` component, and use that key to automatically prefix the `id` prop.
+
+[rn-tourguide](https://github.com/xcarpentier/rn-tourguide?tab=readme-ov-file#using-multiple-tours) uses a similar approach.
+
+```jsx
+// No need to prefix the ids manually in steps array.
+
+<Tour tourKey="global" steps={globalSteps} />;
+<Tour tourKey="paymentlinks" steps={paymentLinksSteps} />;
+
+// file: DashboardPage.tsx
+<TourStep tourKey="global" id="step-1">
+  <Button>Click me</Button>
+</TourStep>;
+
+// file: PaymentLinks.tsx
+<TourStep tourKey="paymentlinks" id="step-1">
+  <Button>Click me</Button>
+</TourStep>;
+```
+
+### Technical Challenge in React Native
```

Comment:
Why is this a challenge? We will wrap it on screen level and that should be fine right?

Reply:
It's still a technical challenge, because if consumer wants to add two different Tours in the same React subtree they will have to wrap 1 tour over another which will cause issues since React.Context will take the nearest context. 

```
<Tour tourKey="global">
  <Tour tourKey="payments">
    <Example1 /> // this will get all the states from the `payments` tour
    <Example2 /> // this will also get all the states from the `payments` tour
  </Tour>
</Tour>
```